### PR TITLE
Experience/10709/navbar redesign

### PR DESCRIPTION
**Pull request by etanb**
**Remaining Navbar Design items (cc @yudonoodles @jillian-hammer):**

- [x] Login button w/blue background has its outline on the outside that makes the size slightly bigger than should be
- [x] Increase space between Login and Connect now buttons to 16px
- [x] Center align the nav items
- [x] Increase padding between submenu item and the bottom of dropdown (playing with .5rem or 1rem)
- [x] Mobile nav bug: active link indicator doesn't appear, esp when toggling between items. if the active link is there, it disappears upon hover.

**Remaining Engineering items:** 

- [x] Integrate new navbar into latest pages
- [x] Utilize older navbar for logged in pages
- [x] Consolidate as much of the override SCSS as possible

**Showcase Gallery:**

<details>
  <summary>Default navbar scheme, logged out</summary>
  
Screenshot 2023-08-29 at 4 06 06 PM

  
</details>

<details>
  <summary>Blue variant navbar scheme, logged out</summary>
  
Screenshot 2023-08-29 at 4 05 56 PM

</details>

<details>
  <summary>Default navbar scheme, logged in Admin state</summary>
  
Screenshot 2023-08-29 at 4 05 35 PM

</details>

<details>
  <summary>Blue variant navbar scheme, logged in Admin state</summary>
  
Screenshot 2023-08-29 at 4 04 45 PM

</details>

<details>
  <summary>Default navbar scheme w/ Dropdown Nav Expanded, logged in Admin state</summary>
  
Screenshot 2023-08-29 at 4 05 44 PM

</details>

<details>
  <summary>Blue variant navbar scheme w/ Dropdown Nav Expanded, logged in Admin state</summary>
  
Screenshot 2023-08-29 at 4 04 23 PM

</details>

<details>
  <summary>MOBILE: Default navbar scheme, logged in Admin state</summary>
  
Screenshot 2023-08-29 at 4 09 51 PM

</details>

<details>
  <summary>MOBILE: Blue variant navbar scheme, logged in Admin state</summary>

Screenshot 2023-08-29 at 4 09 33 PM

</details>

**User stories:**

- A default white Navbar experience.
- A blue variant Navbar experience if `isFullWidth` is `true`.
- Customized dropdown nav items and nav link components with custom hover states for both variants.
- Mobile styling for nav items and dropdowns with custom hover states for both variants.

**Notes:**

- I removed all usage of our old Navbar, ReportStreamHeader.
- Also I removed the new `<CallToAction />` component since it seemed redundant with the logic we have in the `<USLink />` component.
- Refactored the OktaWidget to attach to a specific element so it no longer renders in a `<main></main>` HTML element so we can wrap the entire app's content in a `<main></main>` tag.
- Using `decorators` and mocking session context, I was able to mock different situations in our Storybook story (yay).
- As you can see, there's QUITE A LOT of CSS overrides happening in `ReportStreamNavbar.module.scss` which is one of the deep caveats of highly customizing a pre-built USWDS component. I tried to consolidate them as much as possible.

### CODE REVIEW

File: frontend-react/src/components/header/ReportStreamNavbar.tsx [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+                <a href="/admin/message-tracker" key="message-tracker">
+                    Message Id Search
+                </a>,
+                <a href="/admin/send-dash" key="send-dash">
+                    Receiver Status Dashboard
+                </a>,
+                <a href="/admin/value-sets" key="value-sets">
+                    Value Sets
+                </a>,
+                <a href="/admin/validate" key="validate">
```
Comment (penny-lischer) [inline frontend-react/src/components/header/ReportStreamNavbar.tsx:146]: This link is not working

File: frontend-react/src/components/USLink.test.tsx [hunk 2]
```diff
@@ -162,6 +162,32 @@ describe("USLinkButton", () => {
             );
         },
     );
+
+    test("renders USLinkButton with an icon if it has an external link and extLinkIcon is set", () => {
+        renderApp(
+            <USLinkButton
+                extLinkIcon
+                href="https://app.smartsheetgov.com/b/form/48f580abb9b440549b1a9cf996ba6957"
```
Comment (penny-lischer) [inline frontend-react/src/components/USLink.test.tsx:170]: maybe reference {site.forms.intakeElr.url} incase url becomes obsolete.

File: frontend-react/src/components/header/ReportStreamNavbar.tsx [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+                                        Logout
+                                    </Button>
+                                </>
+                            ) : (
+                                <>
+                                    <USLinkButton href="/login">
+                                        Login
+                                    </USLinkButton>
+                                    <USLinkButton
+                                        href="https://app.smartsheetgov.com/b/form/48f580abb9b440549b1a9cf996ba6957"
```
Comment (penny-lischer) [inline frontend-react/src/components/header/ReportStreamNavbar.tsx:233]: please replace with {site.forms.connectWithRS.url}

File: frontend-react/vite.config.ts.timestamp-1693432977956-f5271dc17152b.mjs
Comment (penny-lischer) [inline frontend-react/vite.config.ts.timestamp-1693432977956-f5271dc17152b.mjs:1]: Was this file intentionally added?

Comment (etanb) [reply frontend-react/vite.config.ts.timestamp-1693432977956-f5271dc17152b.mjs:1]: nope! when a vite process is interrupted, it creates this file. will remove.

**Review by penny-lischer — APPROVED** [review]
LGTM

**Review by jpandersen87 — APPROVED** [review]

**Pull request merged**